### PR TITLE
Devise + Register feature test

### PR DESCRIPTION
**Pull request by Prozak8**
https://www.pivotaltracker.com/story/show/161013303

* Devise added without mail_action
* Feature test for sign up
* Unit test for sign up

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (rupertlion) [inline spec/models/user_spec.rb]: I would add something here for posterity... something like (may need to adjust a bit):

```  describe 'DB table' do
    it { is_expected.to have_db_column :email }
    it { is_expected.to have_db_column :password }
  end

  describe 'Validations' do
    it { is_expected.to validate_presence_of :email }
    it { is_expected.to validate_presence_of :encrypted_password }
  end

  describe 'Factory' do
    it 'should have valid Factory' do
      expect(FactoryBot.create(:user)).to be_valid 
    end
  end```

File: config/initializers/devise.rb [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+  # When using OmniAuth, Devise cannot automatically set OmniAuth path,
+  # so you need to do it manually. For the users scope, it would be:
+  # config.omniauth_path_prefix = '/my_engine/users/auth'
+
+  # ==> Turbolinks configuration
+  # If your app is using Turbolinks, Turbolinks::Controller needs to be included to make redirection work correctly:
+  #
+  # ActiveSupport.on_load(:devise_failure_app) do
+  #   include Turbolinks::Controller
+  # end
```
Comment (angiecstn) [inline config/initializers/devise.rb]: I'd get rid of comments

File: spec/models/user_spec.rb (hunk 1 above)
Comment (Prozak8) [reply spec/models/user_spec.rb]: Good idea. I'll add everything except the Factory. Factory is for a logged in user if I'm not mistaken?

Comment (rupertlion) [reply spec/models/user_spec.rb]: well device creates a user factory for you (see file above), so you will still be using this. the factory spec in this file is just validating that factory exists as an extra step

File: config/initializers/devise.rb (hunk 2 above)
Comment (Prozak8) [reply config/initializers/devise.rb]: Left some in that were regarding OmniAuth. Deleted about 70%

File: spec/models/user_spec.rb (hunk 1 above)
Comment (Prozak8) [reply spec/models/user_spec.rb]: ahhhh alright. Will add it then

**Pull request merged**